### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Ashkanthegreat**
Finished merging everyone's code into refactor and ran tests.  Ready for Master.

### CODE REVIEW

File: lib/calculable.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+  def average(numerator, denominator)
+    (numerator.to_f/denominator).round(2)
+  end
+
+  def calculate_win_percentage(team_games)
+    wins = 0.0
+    team_games.each {|game| wins += 1.0 if game.result == "WIN"}
+    (wins / team_games.length)
+  end
```
Comment (jpc20) [inline lib/calculable.rb:11]: Tests???

File: lib/season_stats.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+require_relative "stats"
+require 'pry'
```
Comment (arpariseau) [inline lib/season_stats.rb:2]: Can pull the pry out here.

File: lib/stats.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  def total_games_and_goals_by_team(team_id, hoa)
+    goals_games = [0, 0]
+    @game_teams.each do |game_team|
+      if hoa && game_team.team_id == team_id && game_team.hoa == hoa
+        add_goals_and_games(goals_games, game_team)
+      elsif !hoa && game_team.team_id == team_id
+        add_goals_and_games(goals_games, game_team)
+      end
+    end
+    goals_games
```
Comment (jpc20) [inline lib/stats.rb:43]: Maybe we can refactor this method?
Let me know if you guys have any ideas.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -16,272 +19,131 @@ def self.from_csv(locations)
...
-    @game_teams = GameTeam.all
-  end
-
-  def percentage_home_wins # move down into game class(class method?)
-    home_wins = @games.find_all {|game| game.home_goals > game.away_goals}
-    (home_wins.length.to_f / @games.length.to_f).round(2) # percentage can be added to a module
+    @game_stats = GameStats.new(Game.all, Team.all, GameTeam.all)
+    @league_stats = LeagueStats.new(Game.all, Team.all, GameTeam.all)
+    @season_stats = SeasonStats.new(Game.all, Team.all, GameTeam.all)
+    @team_stats = TeamStats.new(Game.all, Team.all, GameTeam.all)
```
Comment (arpariseau) [inline lib/stat_tracker.rb:27]: instantiating all four Stats subclasses here

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -16,272 +19,131 @@ def self.from_csv(locations)
 
   def initialize(games_path, teams_path, game_teams_path)
     Game.from_csv(games_path)
+    Team.from_csv(teams_path)
     GameTeam.from_csv(game_teams_path)
```
Comment (arpariseau) [inline lib/stat_tracker.rb:23]: only loading the csv once should save us time

File: lib/league_stats.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+require_relative "stats"
+
+class LeagueStats < Stats
+
+  def count_of_teams
+    @teams.length
+  end
+
+  def best_offense
+    id = unique_team_ids.max_by {|team_id| average_goals_by_team(team_id)}
```
Comment (jpc20) [inline lib/league_stats.rb:10]: Moved these two helpers to the Stats parent class.

File: lib/season_stats.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+class SeasonStats < Stats
+
+  def most_accurate_team(season)
```
Comment (Ashkanthegreat) [inline lib/season_stats.rb:6]: Maybe refactor to clean up?

**Pull request merged**